proto/substrait/type.proto

```diff
@@ -159,6 +163,18 @@ message Type {
     Nullability nullability = 4;
   }
 
+  message PrecisionTimestamp {
+    optional int32 precision = 1;
```

Comment:
I don't know that we've used `optional` much elsewhere.  I seem to recall it being largely a no-op as far as protobuf is concerned.  However, I don't see much harm in it.  Maybe document the default (6 I assume) again?

Reply:
Optional is from the version 2 of the protobuf format.  These days most of the outside world is using version 3 (as is this file) where optional is assumed.